### PR TITLE
Fix : Memory Allocation fixes

### PR DESCRIPTION
**Pull request by urmitchauhan**
## Related Issue

- ACRView retain in the memory that may create some bad access crashes.
- Fix dependancy injection, refactor variable allocation.
- make ACRView weak reference.

Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issues fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one if reasonable to aid 
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

**** | **Before Memory Allocations**    |  **After Memory Allocations**
:-------------------------:|:-------------------------:|:-------------------------:
**Initial allocation** | 1_initial|Screenshot 2022-09-22 at 10 31 29 AM
**Card allocation** | 2_first_render_card|Screenshot 2022-09-22 at 10 02 49 AM
**Remove Card allocation** | 3_first_remove_card|Screenshot 2022-09-22 at 10 03 23 AM

## Description
Create a [descriptive title and provide all of the relevant details in the description](https://chris.beams.io/posts/git-commit/). 
This information lives with the code and will help future readers (***including yourself***) and will serve as documentation.

At the very least please describe the issue you are addressing, and what your fix entails. 

## Sample Card
If appropriate, please include a link to a card in one of the samples directories that can be used to validate this change. This can be an existing card or a card added with this PR.

## How Verified
How you verified the fix, including one or all of the following: 
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it 
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

File: source/macos/AdaptiveCards/AdaptiveCards/Managers/ACSFillerSpaceManager.swift [hunk 1]
```diff
@@ -53,6 +53,11 @@ class ACSFillerSpaceManager {
         stretchableViews = [NSValue]()
         paddingSet = NSHashTable<NSView>(options: .weakMemory, capacity: 5)
         paddingConstraints = [NSLayoutConstraint]()
+        logInfo("init...")
+    }
+    
+    deinit {
+        logInfo("deinit...")
```
Comment (HemDutt) [inline source/macos/AdaptiveCards/AdaptiveCards/Managers/ACSFillerSpaceManager.swift:60]: remove these logs or make them more informative

**Review by sidkotha — APPROVED** [review]

File: source/macos/AdaptiveCards/AdaptiveCards/Managers/ACSVisibilityManager.swift [hunk 2]
```diff
@@ -89,7 +90,10 @@ class ACSVisibilityManager {
     /// change the visibility of the padding of a host view to `visibility`
     /// `isHidden` `YES` indicates that the padding will be hidden
     func changeVisibilityOfPadding(_ hostView: NSView, visibilityHidden isHidden: Bool) {
-        guard let spacerViews = fillerSpaceManager.getFillerSpaceView(hostView) else { return }
+        guard let spacerViews = fillerSpaceManager.getFillerSpaceView(hostView) else {
+            logInfo("couldn't found separator in filler manager..")
```
Comment (mukuagar) [inline source/macos/AdaptiveCards/AdaptiveCards/Managers/ACSVisibilityManager.swift:94]: This log is for padding, not separator

**Review by mukuagar — APPROVED** [review]

File: source/macos/AdaptiveCards/AdaptiveCards/Managers/ACSVisibilityManager.swift (hunk 2 above)
Comment (urmitchauhan) [reply source/macos/AdaptiveCards/AdaptiveCards/Managers/ACSVisibilityManager.swift:94]: yes, will change

Comment (urmitchauhan) [reply source/macos/AdaptiveCards/AdaptiveCards/Managers/ACSVisibilityManager.swift]: Done ✅

**Pull request merged**